src/infra/core/src/utils/datasets_filtering.rs

```diff
@@ -48,3 +60,73 @@ pub fn filter_datasets_by_pattern(
             .boxed()
     }
 }
+
+pub fn filter_datasets_by_any_pattern(
+    dataset_repo: &dyn DatasetRepository,
+    remote_repo_reg: Arc<dyn RemoteRepositoryRegistry>,
+    dataset_ref_patterns: Vec<DatasetRefAnyPattern>,
+) -> Chain<FilteredDatasetRefAnyStream<'_>, FilteredDatasetRefAnyStream<'_>> {
+    let clone_dataset_ref_patterns = dataset_ref_patterns.clone();
+
+    let static_stream = Box::pin(async_stream::try_stream! {
+        for dataset_ref_pattern in &dataset_ref_patterns.clone() {
+            if !dataset_ref_pattern.is_pattern() {
+                yield dataset_ref_pattern
+                    .as_dataset_ref_any()
+                    .unwrap()
+                    .clone();
+            } else if dataset_ref_pattern.is_remote() {
```

Comment:
what happens to local patters like `dataset%` or `acc/dataset%`?

if I'm reading this correctly they will be skipped?